wordpress-mobile/WordPress-Android
# [Dependency Updates] Test Instructions Documentations per Dependency Update

Parent #17551
Internal Discussion `pcdRpT-2Hp-p2`

This PR adds a new documentations page with [test instructions per dependency update](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md).

FYI: The goal is to further automate the dependency update process by documenting all the specific dependency updates that require explicit testing instructions. The hypothesis here is that this approach would not only save time and effort for each update, but also elevate the quality of dependency update work by ensuring that the testing aspect is not left to chance.

PS: This PR is also slightly updating the [pull request template](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/.github/PULL_REQUEST_TEMPLATE.md) by:
1. Splitting each section using a horizontal line, for grouping purposes (38a4d69483d544015bc05df5fb92cbfcd0f62318),
2. Starting each section with a second-level heading, for consistency purposes (6b80fa7282112eb518479a3fe89e7d92b90d128b), and
3. Applying other minor changes, for consistency purposes (273a96f21311094292777482ad257224c194ca83).

-----

@irfano I added you as the main reviewer, not so randomly (you shared lots of good ideas), since I just wanted someone from the Jetpack/WordPress mobile team to be aware and sign-off on that change. Feel free to merge this PR directly yourself if you deem so.

-----

## To Test:

1. Proofread the new [test instructions per dependency update](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md) documentation page.
    - [Purpose](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md#purpose)
    - [FAQ](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md#faq)
    - [Content](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md#content)
    - [Smoke Test](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md#smoke-test)
    - [Special](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md#special)
2. Verify that each section is well written, can be easily understood and provide enough value while devs are working on a specific dependency update.

FYI: Note that the [Smoke Test](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md#smoke-test) and [Special](https://github.com/wordpress-mobile/WordPress-Android/blob/docs/test_instructions_per_dependency_update/docs/test_instructions_per_dependency_update.md#special) sections can be further improved. However, I didn't want make this improvement now and instead added TODO for devs to keep improving the documentation themselves when and if needed.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

4. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To Test` section above.

5. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:


-----

UI Changes testing checklist:
